### PR TITLE
Enhance useStream with default value

### PR DESCRIPTION
**Pull request by michaltaberski**

### CODE REVIEW

**Review by tlaundal — CHANGES_REQUESTED** [review]
Nice idea, could you give a use case example? I'm a bit curious as to how the call site would look with and without this.

Requesting changes because the docstring must be updated and we should add a small test for this.

**Review by tlaundal — APPROVED** [review]
Vacation review on mobile, but LGTM!

**Comment by michaltaberski** [comment]
It's not critical PR, so it can wait for a proper review once you are back. Enjoy!

File: src/react/connect.ts [hunk 1]
```diff
@@ -13,9 +13,15 @@ export type NOT_YET_EMITTED = typeof NOT_YET_EMITTED;
  * has no guarantee for when the first emit will happen.
```
Comment (tlaundal) [inline src/react/connect.ts:13]:
```suggestion
 * has no guarantee for when the first emit will happen. This initial
 * value may be overridden with the `defaultValue` argument.
```

File: src/react/connect.ts [hunk 2]
```diff
@@ -13,9 +13,15 @@ export type NOT_YET_EMITTED = typeof NOT_YET_EMITTED;
...
  *
  * @param source$ Stream that provides the needed values
+ * @param defaultValue Default value returned on init until stream emits new value
  */
-export const useStream = <T>(source$: Observable<T>): T | NOT_YET_EMITTED => {
-  const [value, setValue] = useState<T | NOT_YET_EMITTED>(NOT_YET_EMITTED);
+export const useStream = <T>(
+  source$: Observable<T>,
+  defaultState?: T
```
Comment (tlaundal) [inline src/react/connect.ts:20]:
```suggestion
  defaultValue?: T
```
Keep in line with the `@param` above.

**Review by tlaundal — APPROVED** [review]
:100:

**Pull request merged**